### PR TITLE
7 patch by id

### PR DESCRIPTION
**Pull request by kalumwildig**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -3,6 +3,7 @@ const app = require("../app.js");
 const seed = require("../db/seeds/seed");
 const data = require("../db/data/test-data");
 const db = require("../db/connection");
+const { Test } = require("supertest");
```
Comment (daviidbartlett) [inline __tests__/app.test.js:7]: what this used for?

File: __tests__/app.test.js [hunk 2]
```diff
@@ -64,8 +65,78 @@ describe("GET /api/articles/:article_id", () => {
...
+          article_id: 1,
+          title: "Living in the shadow of a great man",
+          topic: "mitch",
+          author: "butter_bridge",
+          body: "I find this existence challenging",
+          created_at: "2020-07-09T20:11:00.000Z",
+          votes: 88,
+        });
+      });
+  });
```
Comment (daviidbartlett) [inline __tests__/app.test.js:93]: does it work with positive numbers?

File: __tests__/app.test.js [hunk 3]
```diff
@@ -64,8 +65,78 @@ describe("GET /api/articles/:article_id", () => {
...
+//           expect(users).toHaveLength(4);
+//           users.forEach((element) => {
+//             expect(element).toEqual(
+//               expect.objectContaining({
+//                 username: expect.any(String),
+//               })
+//             );
+//           });
+//         });
+//     });
```
Comment (daviidbartlett) [inline __tests__/app.test.js:141]: why is this in this PR?

**Review by daviidbartlett — COMMENTED** [review]
You don't need to check if an id exists on all endpoints!
Please don't use  method `idExist` unnecessarily. Use your tests to drive the code, not the lecture examples.

File: util_funcs.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+exports.idExist = (rows, id) => {
+  const results = rows[0];
+  if (!results) {
+    return Promise.reject({
```
Comment (popatre) [inline util_funcs.js:4]: Does this error message work for articles?

**Review by popatre — CHANGES_REQUESTED** [review]

**Pull request merged**